core-server/src/main/java/org/glassfish/jersey/server/internal/routing/PathMatchingRouter.java

```diff
@@ -80,11 +92,42 @@
             }
         }
 
+        if (result == null && acceptedRouteCandidate != null) {
+            //method designator mismatched, but still go the route to get the proper status code
+            result = matchPathSelected(context, acceptedRouteCandidate, matchResultCandidate, tracingLogger);
+        }
+
         if (result == null) {
             // No match
             return Router.Continuation.of(context);
         }
 
         return result;
     }
+
+    private Router.Continuation matchPathSelected(final RequestProcessingContext context, final Route acceptedRoute,
+                                                  final MatchResult matchResult, final TracingLogger tracingLogger) {
+        // Push match result information and rest of path to match
+        context.routingContext().pushMatchResult(matchResult);
+        final Router.Continuation result = Router.Continuation.of(context, acceptedRoute.next());
+
+        // tracing
+        tracingLogger.log(ServerTraceEvent.MATCH_PATH_SELECTED, acceptedRoute.routingPattern().getRegex());
+
+        return result;
+    }
+
+    /**
+     * Return true iff the method is a sub-resource and not a locator and http method designator does not match
```

Comment:
typo... (iff)

Reply:
iff stands for [if and only if](https://en.wikipedia.org/wiki/If_and_only_if)